cli/deno_dir.rs

```diff
@@ -534,100 +731,95 @@ fn filter_shebang(bytes: Vec<u8>) -> Vec<u8> {
   }
 }
 
-/// Asynchronously fetch remote source file specified by the URL `module_name`
-/// and write it to disk at `filename`.
-fn fetch_remote_source_async(
-  module_name: &str,
-  filename: &str,
-) -> impl Future<Item = Option<ModuleMetaData>, Error = DenoError> {
-  let filename = filename.to_string();
-  let module_name = module_name.to_string();
-  let p = PathBuf::from(filename.clone());
-  // We write a special ".mime" file into the `.deno/deps` directory along side the
-  // cached file, containing just the media type.
-  let media_type_filename = [&filename, ".mime"].concat();
-  let mt = PathBuf::from(&media_type_filename);
-  eprintln!("Downloading {}", &module_name);
-  http_util::fetch_string(&module_name).and_then(
-    move |(source, content_type)| {
-      match p.parent() {
-        Some(ref parent) => fs::create_dir_all(parent),
-        None => Ok(()),
-      }?;
-      deno_fs::write_file(&p, &source, 0o666)?;
-      // Remove possibly existing stale .mime file
-      // may not exist. DON'T unwrap
-      let _ = std::fs::remove_file(&media_type_filename);
-      // Create .mime file only when content type different from extension
-      let resolved_content_type = map_content_type(&p, Some(&content_type));
-      let ext = p
-        .extension()
-        .map(|x| x.to_str().unwrap_or(""))
-        .unwrap_or("");
-      let media_type = extmap(&ext);
-      if media_type == msg::MediaType::Unknown
-        || media_type != resolved_content_type
-      {
-        deno_fs::write_file(&mt, content_type.as_bytes(), 0o666)?
-      }
-      Ok(Some(ModuleMetaData {
-        module_name: module_name.to_string(),
-        filename: filename.to_string(),
-        media_type: map_content_type(&p, Some(&content_type)),
-        source_code: source.as_bytes().to_owned(),
-        maybe_output_code_filename: None,
-        maybe_output_code: None,
-        maybe_source_map_filename: None,
-        maybe_source_map: None,
-      }))
-    },
-  )
+#[derive(Debug)]
+pub struct SourceMetaData {
```

Comment:
Does "Source" in this case mean "source code" or the "remote source"? Choose a name that is clearer.